src/components/navigation/Navigation.jsx

```diff
@@ -4,25 +4,44 @@ import TextField from '@material-ui/core/TextField';
 import Search from '@material-ui/icons/Search';
 import Add from '@material-ui/icons/Add';
 import './Navigation.css';
+import PropTypes from 'prop-types';
 
 export class Navigation extends React.Component {
 
+    constructor(props) {
+        super(props);
+        this.state = { filter: '' };
+    }
+
+    static get propTypes() {
+        return {
+            onAdd: PropTypes.func.isRequired,
```

Comment:
what does another dev can understand from prop `onAdd` or `onFilter`???
what add? etc. `addCourse` more better and meaningful

Please think about naming and give good names to actions and thunks. Not just in that component

Reply:
fixed